platforms/core-runtime/instrumentation-declarations/src/main/java/org/gradle/internal/instrumentation/property/upgrades/PropertyUpgradeInstrumentationRegistry.java

```diff
@@ -24,6 +25,6 @@
  * Classes that should be visited should be added to the @VisitForInstrumentation annotation, e.g.
  * {@code @VisitForInstrumentation(value = {Checkstyle.class})}.
  */
-@VisitForInstrumentation(value = {})
+@VisitForInstrumentation(value = {Checkstyle.class})
```

Comment:
❓ Why do we need this here for `Checkstyle` but not for anything else?

Reply:
Checkstyle is in mixed Groovy/Java source set where we can't use annotation processor. 
So we have to use `@VisitForInstrumentation` in another project, where we can have annotation processor enabled.